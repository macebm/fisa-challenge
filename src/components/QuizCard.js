import { useState } from "react";

export default function QuizCard({ question, answer }) {
  const [showAnswer, setShowAnswer] = useState(false);
  return (
    <div className="extra__quizcard">
      <p>{question}</p>
      <button className="extra__btn" onClick={() => setShowAnswer(!showAnswer)}>Toggle answer</button>
      {showAnswer && <p>{answer}</p>}
    </div>
  );
}

