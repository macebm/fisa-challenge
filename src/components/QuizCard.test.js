import QuizCard from "./QuizCard";
import { render, fireEvent } from "@testing-library/react";

describe("QuizCard", () => {
  it("renders correctly", () => {
    const { debug, container } = render(
      <QuizCard question="Foo?" answer="Bar!" />
    );
    debug();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("shows the question and answer texts", () => {
    const { getByText, queryByText, getByRole } = render(
      <QuizCard question="Foo?" answer="Bar!" />
    );
    expect(getByText("Foo?")).toBeInTheDocument();

    expect(queryByText("Bar!", { exact: false })).not.toBeInTheDocument();
    const button = getByRole("button", { name: "Toggle answer" });
    fireEvent.click(button);
    expect(queryByText(/bar/i)).toBeInTheDocument();
  });
});
