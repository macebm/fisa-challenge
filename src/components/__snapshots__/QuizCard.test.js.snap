// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizCard renders correctly 1`] = `
<div
  class="extra__quizcard"
>
  <p>
    Foo?
  </p>
  <button
    class="extra__btn"
  >
    Toggle answer
  </button>
</div>
`;
